### PR TITLE
Add Test Suite to github workflow

### LINKED ISSUE #211: [SURVEY] Important Test Cases
I'd like to know core test cases to add on build workflows like below in order to check the riskiness of PR.
Please leave comments on important test cases, for example `common/sigmgr`.

image

### PR DESCRIPTION
**Pull request by tdrozdovsky**
Signed-off-by: Taras Drozdovskyi <t.drozdovsky@samsung.com>

# Description

During the launch of a new PR or `push` command, a test suite (unit testing) will be launched.
**It should be noted that testing for the following packages (`src/controller/discoverymgr src/controller/servicemgr/executor/containerexecutor src/orchestrationapi`) is temporary excluded due to non-working tests!**

Fixes #211  

## Type of change

- [x] Code cleanup/refactoring

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: .github/workflows/edge-home-orchestration-go.yml [hunk 1]
```diff
@@ -8,5 +8,20 @@ jobs:
...
+          go get github.com/axw/gocov/gocov
+          echo "$HOME/go/bin" >> $GITHUB_PATH
+  
       - name: Build the project
         run: ./build.sh
+
+      - name: Run the Test Suite
+        run: |
+          docker stop edge-orchestration
+          gocov test -v $(go list ./src/... | grep -v src/controller/discoverymgr | grep -v src/controller/servicemgr/executor/containerexecutor | grep -v src/orchestrationapi) -coverprofile=/dev/null
```
Comment (t25kim) [inline .github/workflows/edge-home-orchestration-go.yml:27]: How about omitting -v since it doesn't need full logs?

**Review by t25kim — COMMENTED** [review]
Thanks for the valuable PR.
Please take a look my comments.

I found some messages at [build summary](https://github.com/lf-edge/edge-home-orchestration-go/actions/runs/494517250) derived from t.Log() of [types_test.go](https://github.com/lf-edge/edge-home-orchestration-go/blob/f97a8222b745ca0951b7abdcb50797b1990f0437/src/controller/servicemgr/types_test.go#L62)  and [errormsg_test.go](https://github.com/lf-edge/edge-home-orchestration-go/blob/f97a8222b745ca0951b7abdcb50797b1990f0437/src/common/errormsg/errormsg_test.go#L42). 
Do you think they should be fixed or not?

With  `root` privilege, some TCs(such as common/logmgr, controller/securemgr/authenticator) got failed in my ubuntu PC. Can you help to check they should be fixed?

File: .github/workflows/edge-home-orchestration-go.yml (hunk 1 above)
Comment (tdrozdovsky) [reply .github/workflows/edge-home-orchestration-go.yml]: Done!

**Comment by tdrozdovsky** [comment]
> I found some messages at [build summary](https://github.com/lf-edge/edge-home-orchestration-go/actions/runs/494517250) derived from t.Log() of [types_test.go](https://github.com/lf-edge/edge-home-orchestration-go/blob/f97a8222b745ca0951b7abdcb50797b1990f0437/src/controller/servicemgr/types_test.go#L62) and [errormsg_test.go](https://github.com/lf-edge/edge-home-orchestration-go/blob/f97a8222b745ca0951b7abdcb50797b1990f0437/src/common/errormsg/errormsg_test.go#L42).
> Do you think they should be fixed or not?
 
I think it can be removed. t.Log() will not show up until after the test is complete, so if you're trying to debug a test that is hanging or performing badly it seems you need to use fmt.

**Comment by tdrozdovsky** [comment]
> With `root` privilege, some TCs(such as common/logmgr, controller/securemgr/authenticator) got failed in my ubuntu PC. Can you help to check they should be fixed?

@t25kim Please describe how you test with root and I will try to help with this.

**Review by t25kim — APPROVED** [review]
LGTM. Thanks for the great contribution!

**Review by MoonkiHong — APPROVED** [review]
LGTM.

**Pull request merged**